docs/development-guide.md

```diff
@@ -1,128 +1,158 @@
+```
 
-    **Troubleshooting:** If you get "No space left on device" error on Google
-    Cloud Shell, you can build the images on Google Cloud Build: [Enable the
-    Cloud Build
-    API](https://console.cloud.google.com/flows/enableapi?apiid=cloudbuild.googleapis.com),
-    then run `skaffold run -p gcb --default-repo=gcr.io/[PROJECT_ID]` instead.
+```sh
+gcloud auth configure-docker -q
+```
 
-4.  Find the IP address of your application, then visit the application on your
-    browser to confirm installation.
+#### 3.  In the root of this repository, run `skaffold run --default-repo=gcr.io/[PROJECT_ID]`, where [PROJECT_ID] is your GCP project ID.
 
-        kubectl get service frontend-external
+This command:
 
+- Builds the container images
+- Pushes them to GCR
+- Applies the `./kubernetes-manifests` deploying the application to Kubernetes.
 
-## Option 2 - Local Cluster 
+> 🎯 &nbsp;&nbsp;**Troubleshooting:** If you get `"No space left on device"` error on Google
+Cloud Shell, you can build the images on Google Cloud Build: [Enable the
+Cloud Build
+API](https://console.cloud.google.com/flows/enableapi?apiid=cloudbuild.googleapis.com),
+then run `skaffold run -p gcb --default-repo=gcr.io/[PROJECT_ID]` instead.
 
-1. Launch a local Kubernetes cluster with one of the following tools:
+#### 4.  Find the IP address of your application, then visit the application on your browser to confirm installation.
+```sh
+kubectl get service frontend-external
+```
+---
 
-    - To launch **Minikube** (tested with Ubuntu Linux). Please, ensure that the
-       local Kubernetes cluster has at least:
-        - 4 CPUs
-        - 4.0 GiB memory
-        - 32 GB disk space
+## Option 2 - Local Cluster
 
-      ```shell
-      minikube start --cpus=4 --memory 4096 --disk-size 32g
-      ```
+#### 1. A local Kubernetes cluster can be launched with one of the following tools:
 
-    - To launch **Docker for Desktop** (tested with Mac/Windows). Go to Preferences:
-        - choose “Enable Kubernetes”,
-        - set CPUs to at least 3, and Memory to at least 6.0 GiB
-        - on the "Disk" tab, set at least 32 GB disk space
+- **Minikube** (tested with Ubuntu Linux).
+  - Please, ensure that the local Kubernetes cluster has at least:
+    - 4 CPUs
+    - 4.0 GiB memory
+    - 32 GB disk space
 
-    - To launch a **Kind** cluster:
+    ```shell
+    minikube start --cpus=4 --memory 4096 --disk-size 32g
+    ```
 
-      ```shell
-      kind create cluster
-      ```
+- **Docker for Desktop** (tested with Mac/Windows).
+  - Go to Preferences:
+    - Choose “Enable Kubernetes”,
+    - Set CPUs to at least 3, and Memory to at least 6.0 GiB
+    - On the "Disk" tab, set at least 32 GB disk space
 
-2. Run `kubectl get nodes` to verify you're connected to the respective control plane.
+- **Kind** cluster:
+  - Use the basic cluster configuration provided at `local/kind-cluster.yaml`
+    ```shell
+    kind create cluster --config ./local/kind-cluster.yaml
```

Comment:
I don't think this YAML should live in the local dir.  because local will support minikube too

Reply:
Moved it to a new folder `extras`